src/heartratetask/HeartRateTask.cpp

```diff
@@ -110,10 +82,174 @@ void HeartRateTask::StartMeasurement() {
   heartRateSensor.Enable();
   ppg.Reset(true);
   vTaskDelay(100);
+  measurementStart = xTaskGetTickCount();
 }
 
 void HeartRateTask::StopMeasurement() {
   heartRateSensor.Disable();
   ppg.Reset(true);
   vTaskDelay(100);
 }
+
+void HeartRateTask::StartWaiting() {
+  StopMeasurement();
+  backgroundWaitingStart = xTaskGetTickCount();
+}
+
+void HeartRateTask::HandleGoToSleep() {
+  switch (state) {
+    case States::Running:
+      state = States::Idle;
+      break;
+    case States::Measuring:
+      state = States::BackgroundWaiting;
+      StartWaiting();
```

Comment:
```suggestion
    case States::Measuring:
      state = States::BackgroundMeasuring;
```

I would switch to the measuring part directly, as otherwise we will have a gap for 10s in case continuous measuring is active.

Reply:
That makes sense. Thank you!